### ISSUE TITLE
EKS: Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.

### ISSUE THREAD

**Issue by dhoeld** [aws/amazon-vpc-cni-k8s#270]
### Description:

We are using EKS for running our K8s cluster. At any start of an aws-node pod, there is an ERROR on the logs. Though the pod networking seems to work. Do we have to add any parameter in order to get rid of this ERROR in the logs?

### Setup Information:

* Version of EKS:  v1.11.5-eks-6bad6d
* Version of amazon-vpc-cni-k8s: 1.2.1 and 1.3.0 (both are affected)

### Logs:
```
=====Starting installing AWS-CNI =========
=====Starting amazon-k8s-agent ===========
ERROR: logging before flag.Parse: W1217 10:37:25.707719      13 client_config.go:533] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
```

Thanks!

**Comment by ewbankkit** [comment]
@dhoeld The `Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.` message is a standard message from the [Kubernetes Go client](https://github.com/kubernetes/client-go/blob/6924ba6dfc024771f18d50efebb76e0094c01aa3/tools/clientcmd/client_config.go#L549). It indicates that the client will use environment variables set via kubelet to determine the Kubernetes API server URL; I've never seen it not work 😄.
The `ERROR: logging before flag.Parse:` message comes from the [glog logging library](https://github.com/golang/glog/blob/23def4e6c14b4da8ac2ed8007337bc5eb5007998/glog.go#L680) that `client-go` uses.
The [documentation](https://godoc.org/github.com/golang/glog) explictly says
> flag.Parse must be called before any logging is done

A solution would be to add an explicit call to `flag.Parse()` to [`main.go`](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/main.go).

**Comment by edgan** [comment]
This also doesn't work on 1.1 and there hasn't been a new release. Can we get a new release for this?

**Comment by pc-rshetty** [comment]
i still see this issue even on amazon-k8s-cni:v1.3.3

**Comment by nak3** [comment]
This patch https://github.com/aws/amazon-vpc-cni-k8s/pull/293 should fix the issue, but it looks like [v1.3.3](https://github.com/aws/amazon-vpc-cni-k8s/commits/v1.3.3) did not include it.
